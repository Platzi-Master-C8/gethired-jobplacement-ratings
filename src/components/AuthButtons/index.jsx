import React from 'react';

import Button from '@mui/material/Button';

import { goToEnterprise } from '../../utils';

export const AuthenticationButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Log in
        </Button>
    );
};

export const SignupButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Sign up
        </Button>
    );
};
