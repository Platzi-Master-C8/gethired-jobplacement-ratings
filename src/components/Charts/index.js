import RadarChart from "./RadarChart";
import LineChart from "./LineChart";
export { RadarChart, LineChart }