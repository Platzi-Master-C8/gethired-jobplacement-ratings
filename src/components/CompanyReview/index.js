import CompanyReview from './CompanyReview';

export { CompanyReview };
