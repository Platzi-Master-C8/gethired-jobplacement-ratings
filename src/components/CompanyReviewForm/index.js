import CompanyReviewForm from './CompanyReviewForm';

export { CompanyReviewForm };
