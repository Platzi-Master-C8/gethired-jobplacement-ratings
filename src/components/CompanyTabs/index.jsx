import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Tabs, Tab, Box } from '@mui/material';

const TabPanel = ({ children, value, index }) => {
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
        >
            {value === index && <Box sx={{ p: 3 }}>{children}</Box>}
        </div>
    );
};

TabPanel.propTypes = {
    children: PropTypes.node.isRequired,
    index: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
};

const CompanyTabs = ({ tabsOptions }) => {
    const [value, setValue] = useState(1);

    const handleChange = (_event, newValue) => {
        setValue(newValue);
    };

    return (
        <Box sx={{ width: '100%' }}>
            <Box>
                <Tabs value={value} onChange={handleChange}>
                    {tabsOptions.map((option) => (
                        <Tab key={`TabButton-${option.tabKey}`} label={option.tabKey} />
                    ))}
                </Tabs>
            </Box>
            {tabsOptions.map((option, i) => (
                <TabPanel key={`TabContent-${option.tabKey}`} value={value} index={i}>
                    {option.tabContent}
                </TabPanel>
            ))}
        </Box>
    );
};

CompanyTabs.propTypes = {
    tabsOptions: PropTypes.arrayOf(
        PropTypes.shape({
            tabKey: PropTypes.string.isRequired,
            tabContent: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
        }),
    ).isRequired,
};

export default CompanyTabs;
