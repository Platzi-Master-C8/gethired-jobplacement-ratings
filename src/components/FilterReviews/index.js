import FilterReviews from './FilterReviews';

export {
    FilterReviews
}