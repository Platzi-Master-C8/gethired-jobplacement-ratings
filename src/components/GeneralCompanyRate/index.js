import GeneralCompanyRate from './GeneralCompanyRate';

export { GeneralCompanyRate };
