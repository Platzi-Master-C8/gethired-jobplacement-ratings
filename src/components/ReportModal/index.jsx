import React, { useState } from 'react';
import { styled } from '@mui/material/styles';
import PropTypes from 'prop-types';
import {
    Box,
    Fade,
    Modal,
    Select,
    Button,
    MenuItem,
    TextField,
    Typography,
    InputLabel,
    FormControl,
} from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import LoadingButton from '@mui/lab/LoadingButton';
import { UtilityModal } from '../UtilityModal';
import config from '../../config';

const ReportModalStyle = {
    p: 4,
    top: '50%',
    left: '50%',
    // width: 400,
    boxShadow: 24,
    minWidth: '320px',
    borderRadius: '6px',
    position: 'absolute',
    border: '1px solid #555BFF',
    bgcolor: 'background.paper',
    transform: 'translate(-50%, -50%)',
};

const FormModalStyle = {
    gap: '20px',
    display: 'grid',
};

const initialState = {
    email: '',
    reason: '',
    description: '',
};

const LengthMessage = styled(Box)`
    width: 90%;
    margin-top: 8px;
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const ReportModal = ({ open, handleClose, review_id, reasons }) => {
    const [form, setForm] = useState(initialState);
    const [error, setError] = useState(false);
    const [success, setSuccess] = useState(false);
    const [loading, setLoading] = useState(false);
    const [formError, setFormError] = useState(false);
    const [lengthError, setLengthError] = useState('');

    const handleInput = (e) => {
        const { name, value } = e.target;
        setForm({ ...form, [name]: value });
    };

    const handleReportModal = () => {
        if (error) {
            setError(false);
            return;
        }
        setError(false);
        setSuccess(false);
        handleClose();
    };

    const validateForm = () => {
        if (form.description.length < 10) {
            setLengthError('Field should be greater than 10');
            setLoading(false);
            return false;
        }
        if (form.description.length > 70) {
            setLengthError('Field should be less than 70');
            setLoading(false);
            return false;
        }
        if (!form.reason || !form.email || !form.description) {
            setFormError(true);
            setLoading(false);
            return false;
        }
        setFormError(false);
        setLengthError('');
        return true;
    };

    const handleSubmit = async (e) => {
        e.preventDefault();
        setLoading(true);
        if (!validateForm()) return;
        const url = `${config.api}company-evaluation/${review_id}/complaints`;
        const body = {
            email: form.email,
            problem_description: form.description,
            reporting_reason_type_id: form.reason,
        };
        fetch(url, {
            method: 'POST',
            headers: config.headers,
            body: JSON.stringify(body),
        })
            .then((res) => res.json())
            .then((data) => {
                if (data.id) setSuccess(true);
                else setError(true);
                setForm(initialState);
            })
            .catch(() => setError(true))
            .finally(() => setLoading(false));
    };

    return (
        <Modal open={open} onClose={handleReportModal}>
            <Box sx={ReportModalStyle}>
                <Box sx={{ display: 'flex', justifyContent: 'flex-end' }}>
                    <Button title="Close modal" onClick={handleReportModal} aria-label="Close modal">
                        <CloseIcon />
                    </Button>
                </Box>
                <Box mb="20px">
                    <Typography variant="h6" component="h6">
                        Help us understand what is happening
                    </Typography>
                    <Fade in={formError} style={{ height: formError ? 'initial' : 0 }}>
                        <Typography variant="body1" component="p" color="error">
                            Complete the form to continue
                        </Typography>
                    </Fade>
                </Box>
                <form onSubmit={handleSubmit} style={FormModalStyle}>
                    <FormControl fullWidth>
                        <InputLabel>Select a reason</InputLabel>
                        <Select
                            name="reason"
                            required
                            value={form.reason}
                            onChange={handleInput}
                            label="Select a reason"
                        >
                            <MenuItem value="">None</MenuItem>
                            {reasons?.map((item) => (
                                <MenuItem key={`Reason-${item.id}`} value={item.id}>
                                    {item.name}
                                </MenuItem>
                            ))}
                        </Select>
                    </FormControl>
                    <Box sx={{ display: 'grid', gap: '8px' }}>
                        <Typography>What was the issue with this review?</Typography>
                        <TextField
                            rows={5}
                            required
                            fullWidth
                            multiline
                            id="User-reason"
                            name="description"
                            variant="outlined"
                            onChange={handleInput}
                            value={form.description}
                            label="Describe the issue"
                        />
                        <LengthMessage>
                            <Typography variant="caption" sx={{ color: 'rgb(255, 117, 117)' }}>
                                {lengthError || ''}
                            </Typography>
                            <Typography variant="caption">{form?.description?.length || 0}</Typography>
                        </LengthMessage>
                    </Box>
                    <Box sx={{ display: 'grid', gap: '8px' }}>
                        <Typography>Your email address *</Typography>
                        <TextField
                            required
                            fullWidth
                            type="email"
                            name="email"
                            label="Email"
                            id="User-email"
                            variant="outlined"
                            value={form.email}
                            onChange={handleInput}
                        />
                    </Box>
                    {loading && (
                        <LoadingButton loading loadingIndicator="Loading..." variant="outlined">
                            Fetch data
                        </LoadingButton>
                    )}
                    {!loading && (
                        <Button type="submit" variant="contained" fullWidth size="large">
                            Submit
                        </Button>
                    )}
                </form>
                <UtilityModal
                    open={success || error}
                    handleClose={handleReportModal}
                    message={
                        success
                            ? 'The report was sent correctly, thank you for your time'
                            : 'An error has occurred, please try again'
                    }
                />
            </Box>
        </Modal>
    );
};

ReportModal.propTypes = {
    open: PropTypes.bool.isRequired,
    handleClose: PropTypes.func.isRequired,
    review_id: PropTypes.number.isRequired,
    reasons: PropTypes.arrayOf(
        PropTypes.shape({
            name: PropTypes.string.isRequired,
            id: PropTypes.number.isRequired,
        }),
    ).isRequired,
};

export default ReportModal;
