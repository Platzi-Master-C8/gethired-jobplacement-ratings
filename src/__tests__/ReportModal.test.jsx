import React from 'react';
import { render, fireEvent, within } from '@testing-library/react';
import '@testing-library/jest-dom';

import { ReportModal } from '../components/ReportModal';

const defaultProps = {
    open: true,
    handleClose: () => {},
};

const renderComponent = (props = {}) => render(<ReportModal {...defaultProps} {...props} />);

describe('<ReportModal />', () => {
    test('Component render', () => {
        const { asFragment } = renderComponent({});
        expect(asFragment()).toMatchSnapshot();
    });

    test('Modal fill form', () => {
        const { getByText, getByRole, getByLabelText } = renderComponent({});

        const reasonInput = getByLabelText(/describe the issue/i);
        const emailInput = getByLabelText(/email/i);

        fireEvent.mouseDown(getByRole('button', { expanded: false }));
        const listbox = within(getByRole('listbox'));

        fireEvent.click(listbox.getByText(/Suspicious, spam or fake/i));
        fireEvent.change(reasonInput, { target: { value: 'Example test' } });
        fireEvent.change(emailInput, { target: { value: 'example@test.com' } });
        fireEvent.click(getByText(/submit/i));
    });

    test('it calls the close function', () => {
        const handleClose = jest.fn();
        const { getByTitle } = renderComponent({ handleClose });

        fireEvent.click(getByTitle('Close modal'));
        expect(handleClose).toBeCalled();
    });
});
