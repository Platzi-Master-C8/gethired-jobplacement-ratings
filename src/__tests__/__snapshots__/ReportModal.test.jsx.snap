// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportModal /> Component render 1`] = `<DocumentFragment />`;
